src/authorizer.js

```diff
@@ -46,6 +45,7 @@ class Authorizer {
   }
 
   async ensureApiKey(clientId) {
+    const aws = require('aws-sdk');
```

Comment:
Thinking about this in the context of lambda function:

When we have the `require` in the method body, it is going to be called N times during the lifetime of a lambda container. Before, it was called only once, during the container initialisation. As this is used as the Authorizer function, minimising the response time is important (although, the response is going to be cached).

You could still define the `aws` variable outside of the method, and initialise it only when it is not defined:
```
let aws;
...
async ensureApiKey(clientId) {
  if (!aws) {
    aws = require('aws-sdk');
  }
...
```

Reply:
Node uses a module cache to prevent reloading a whole module multiple times. The question then becomes is it faster to load the data from storing an additional reference in memory, or from a dictionary cache.

We could do all of this outside of the method so it will only be executed once, or even use the  `require.resolve()`, but all the options end up being similar.

Reply:
I was thinking there is some optimization done by node.js already, I should have looked into that. Thanks for the detailed explanation.

Based on that, I am going to resolve the discussion and not request any change here.